addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -25,3 +30,202 @@ def _compute_proxy_type(self):
         for user in self:
             if user.company_id.country_code == 'IT':
                 user.proxy_type = 'l10n_it_edi'
+
+    def _get_proxy_identification(self, company, proxy_type):
+        if proxy_type == 'l10n_it_edi':
+            if not company.l10n_it_codice_fiscale:
+                raise UserError(_('Please fill your codice fiscale to be able to receive invoices from FatturaPA'))
```

Comment:
It's weird to have an UserError in a getter.

Reply:
This isn't a 'getter' that is meant to be used anywhere in the code - it's an overridable interface exposed by AccountEdiProxyClientUser to enable child classes to specify the identification key which the ProxyUser should be registered with on the IAP server.

If the interface expects a UserError to be raised when no identification key can be provided, there's nothing wrong with that IMO.